File: .files/bin/pingtcp.pl
#!/usr/bin/env perl

use strict;
use warnings;
use IO::Socket;

usage() if $#ARGV < 0 ;

foreach (@ARGV) {
  my ($host, $port) = split(/:/); 

  my $remote = IO::Socket::INET->new(
    Proto    => "tcp",
    PeerAddr  => $host,
    PeerPort  => $port,
    Timeout    => 8,
  );
  
  if ($remote) {
    print "connection succeeded: ${host}:${port}\n";
    close $remote;
  }
  else {
    print "connection failed: ${host}:${port} ($@)\n";
  }

}

sub usage {
  print STDERR "USAGE: $0 host:port [ host:port ... ]\n";
  exit 1;
}

File: .files/bin/pingtcp.py
#!/usr/bin/env python2

# python version of pingtcp - makes tcp connection to specified target(s)

import sys, traceback, socket
socket.setdefaulttimeout(5)

try:
    for arg in sys.argv[1:]:

        host, port = arg.split(':',2)
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

        try:
            s.connect((host, int(port)))
            print('connection succeeded: %s:%s' % (host, port))
            s.close()

        except socket.error, e:
            print "connection failed: %s:%s (%s)" % (host, port, e)

except KeyboardInterrupt:
    sys.exit(0)

